SetupDataPkg/Tools/KnobService.py

```diff
@@ -315,37 +334,38 @@ def generate_public_header(schema, header_path, efi_type=False):
         out.write("" + get_line_ending(efi_type))
         out.write("// Schema-defined enums" + get_line_ending(efi_type))
         for enum in schema.enums:
-            if enum.help != "":
-                out.write("// {}".format(enum.help) + get_line_ending(efi_type))
-            out.write("typedef enum {" + get_line_ending(efi_type))
-            has_negative = False
-            for value in enum.values:
-                if value.help != "":
-                    out.write(get_spacing_string(efi_type) + "{}_{} = {}, // {}".format(
-                        enum.name,
-                        value.name,
-                        value.number,
-                        value.help
+            if enum.headerRef == "":

```

Comment:
An idea to consider:

Instead of omitting the definition, it might be useful to emit the output, but wrap it in a #if

This gives the consumer the ability to generate a standalone service/tool if needed, and it also provides more debug output that would allow an engineer to compare the locally generated struct to the struct that is included. These *should* be the same, but this would give the output needed to verify this if there are problems.

i.e.

```c
/* Schema-referenced headers */
#ifdef USE_HEADER_REF_INCLUDES
#include "bar.h" // for bar_options_t
#include "feature.h" // for feature_config_t, fooenum
#endif // #ifdef USE_HEADER_REF_INCLUDES
/*--------------------------*/

...

#ifndef USE_HEADER_REF_INCLUDES
typedef struct bar_options_t {
};
...
#endif // #ifndef USE_HEADER_REF_INCLUDES

#ifndef USE_HEADER_REF_INCLUDES
typedef struct feature_config_t{
};
...
#endif // #ifndef USE_HEADER_REF_INCLUDES

...
```
It might be better to negate the logic so that the includes are the default.

Reply:
The logic was indeed implemented as flipped from this comment. Users can optionally -DSTANDALONE_HEADER_NO_REFS to optionally use the generated file as a standalone header.